cpu/esp32/doc.txt

```diff
@@ -0,0 +1,1144 @@
+```
+### <a name="esp32_flashing_using_docker"> Make Process with Docker Image </a> &nbsp;[[TOC](#esp32_toc)]
+
+Using Docker, the make process consists of the following two steps:
+
+1. **making** the RIOT binary **within a RIOT Docker image**
+2. **flashing** the RIOT binary using a flasher program **on the host system**
+
+Once the RIOT Docker image has been started from RIOT's root directory, a RIOT application can be compiled inside the Docker using the make command as usual, for example:
+
+```
+make BOARD=esp32-esp-12x -C tests/shell ...
+```
+This will generate a RIOT binary in ELF format.
+
+@note You can't use the ```flash``` target inside the Docker image.
+
+The RIOT binary has to be flash outside docker on the host system. Since the Docker image was stared while in RIOT's root directory, the output of the compilations is also accessible on the host system. On the host system, the ```flash-only``` target can then be used to flash the binary.
+```
+make flash-only BOARD=esp32-esp-12x -C tests/shell
+```
+
+
+
+## <a name="esp32_manual_toolchain_installation"> Manual Toolchain Installation </a> &nbsp;[[TOC](#esp32_toc)]
+
+A more difficult way to install the toolchain is the manual installation of required components as described below.
+
+@note To use the precompiled toolchain the following packages (Debian/Ubuntu) have to be installed:<br> ```build-essential``` ```cppcheck``` ```coccinelle``` ```curl``` ```doxygen``` ```git``` ```graphviz``` ```make``` ```pcregrep``` ```python``` ```python-serial``` ```python3``` ```python3-flake8``` ```unzip``` ```wget```
+
+### <a name="esp32_installation_of_xtensa_gcc"> Installation of Xtensa GCC compiler suite </a> &nbsp;[[TOC](#esp32_toc)]
+
+Xtensa GCC compiler for ESP32 can be downloaded and installed as precompiled binary archive from GitHub.
+
+```
+mkdir -p $HOME/esp
+cd $HOME/esp
+git clone https://github.com/gschorcht/xtensa-esp32-elf.git
+```
+Once the compiler is installed you can add the binary directory to your ```PATH``` variable.
+```
+export PATH=$PATH:$HOME/esp/xtensa-esp32-elf/bin
+```
+
+### <a name="esp32_installation_of_esp_idf"> Installation of ESP-IDF (Espressif IoT Development Framework) </a> &nbsp;[[TOC](#esp32_toc)]
+
+ESP-IDF, the official SDK from Espressif, can be downloaded and installed as GIT repository.
+
+```
+cd $HOME/esp
+git clone --recursive https://github.com/espressif/esp-idf.git
+cd esp-idf
+git checkout -q f198339ec09e90666150672884535802304d23ec
+cd components/esp32/lib
+git checkout -q 534a9b14101af90231d40a4f94924d67bc848d5f
+```
+
+@note Please take care to checkout correct branches which were used for the port. Newer versions might also work but were not tested.
+
+Since we only use a few header files, ESP-IDF does not need to be compiled in any way. To use the installed ESP-IDF, just set the variable ```SDK_DIR``` accordingly.
+```
+export SDK_DIR=$HOME/esp/esp-idf
```

Comment:
Should this be `ESP32_SDK_DIR=...` instead `SDK_DIR`

Reply:
Many thanks. Of course, you are right. I had changed it when @kaspar030 asked to change SDK_DIR to ESP_SDK_DIR in the ESP8266 port. Unfortunately, I forgot to change the documentation, as it often happens :worried: